apps/deploy_monitoring/src/monitors/resource.rs

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) SimpleStaking and Tezedge Contributors
+// SPDX-License-Identifier: MIT
+#![forbid(unsafe_code)]
+
+use crate::monitors::Alerts;
+use crate::node::OcamlNode;
+use crate::slack::SlackServer;
+use chrono::Utc;
+use std::collections::HashMap;
+
+use merge::Merge;
+use serde::Serialize;
+use slog::{error, Logger};
+use std::collections::VecDeque;
+use std::sync::{Arc, RwLock};
+
+use getset::Getters;
+use sysinfo::{System, SystemExt};
+
+use shell::stats::memory::ProcessMemoryStatsMaxMerge;
+
+use crate::display_info::DiskData;
+use crate::node::{Node, TezedgeNode, OCAML_PORT, TEZEDGE_PORT};
+
+pub type ResourceUtilizationStorage = Arc<RwLock<VecDeque<ResourceUtilization>>>;
+pub type ResourceUtilizationStorageMap = HashMap<&'static str, ResourceUtilizationStorage>;
+
+/// The max capacity of the VecDeque holding the measurements
+pub const MEASUREMENTS_MAX_CAPACITY: usize = 40320;
+
+pub struct ResourceMonitor {
+    resource_utilization: ResourceUtilizationStorageMap,
+    last_checked_head_level: Option<u64>,
+    alerts: Alerts,
+    log: Logger,
+    slack: Option<SlackServer>,
+    system: System,
+}
+
+#[derive(Clone, Debug, Serialize, Getters, Merge, Default)]
+pub struct MemoryStats {
+    #[get = "pub(crate)"]
+    #[merge(strategy = merge::ord::max)]
+    node: ProcessMemoryStatsMaxMerge,
+
+    #[get = "pub(crate)"]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[merge(strategy = merge::ord::max)]
+    protocol_runners: Option<ProcessMemoryStatsMaxMerge>,
```

Comment:
@adonagy 
please add here `TODO: TE-499 remove protocol_runners and use validators for ocaml and tezedge type`